### PR TITLE
Teacher Application

### PR DESCRIPTION
**Pull request by kvsteele**

### CODE REVIEW

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 1]
```diff
@@ -143,6 +142,7 @@ class StudentForm extends React.Component {
         last_name: this.state.last_name,
         city: this.state.city,
         gender: this.state.gender,
+
```
Comment (sonia-y) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: take out these spaces

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 2]
```diff
@@ -160,6 +160,7 @@ class StudentForm extends React.Component {
         zipcode: this.state.zipcode,
         location_preference: this.state.location_preference,
         travel_distance: this.state.travel_distance,
+
```
Comment (sonia-y) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: same

Comment (sonia-y) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: i don't think you should be editing student form, this page is fine

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 3]
```diff
@@ -1,152 +1,684 @@
...
+    var params = {
+      teacher: {
+        email: this.state.email,
+        password: this.state.password,
+        password_confirmation: this.state.password_confirmation,
+        first_name: this.state.first_name,
+        last_name: this.state.last_name,
+        city: this.state.city,
+        gender: this.state.gender,
+
```
Comment (sonia-y) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: take out spaces

File: app/assets/javascripts/components/form/teacher_form.es6.jsx [hunk 4]
```diff
@@ -1,152 +1,684 @@
...
+        criminal_charges: this.state.criminal_charges,
+        youth_participation: this.state.youth_participation,
+        criminal_explanation: this.state.criminal_explanation,
+        waiver_signature: this.state.waiver_signature,
+        waiver_date: this.state.waiver_date,
+      }
+    };
+    Requester.post(
+      RouteConstants.authentication.signup.teacher,
```
Comment (kvsteele) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: fixed

File: app/assets/javascripts/components/form/teacher_form.es6.jsx (hunk 3 above)
Comment (kvsteele) [inline app/assets/javascripts/components/form/teacher_form.es6.jsx]: fixed

File: app/assets/javascripts/components/form/student_form.es6.jsx [hunk 5]
```diff
@@ -370,6 +370,7 @@ class StudentForm extends React.Component {
                   onChange={(event) => this.handleChange(event)}/>
               </FormGroup>
 
+              {/*Waiting on this one
```
Comment (sonia-y) [inline app/assets/javascripts/components/form/student_form.es6.jsx]: uncomment these

**Pull request merged**